test_files/abstract/abstract.tsickle.ts

```diff
@@ -8,43 +8,61 @@
  * @abstract
  */
 abstract class Base {
+  abstract simple(): void;
+public abstract publicAbstract(): void;
+public abstract params(x: number[]): void;
+
+  // Verify we properly handle functions without a declared return type.
+  abstract noReturnType();
+
+  // Verify we properly handle functions that expect a return value.
+  abstract hasReturnType(): number;
+/**
+ * @return {void}
+ */
+bar() {
+    this.simple();
+    this.publicAbstract();
+    this.params([]);
+    this.noReturnType();
+    this.hasReturnType();
+  }
+}
+
+function Base_tsickle_Closure_declarations() {
```

Comment:
Actually, let me retract my statement. The `@__PURE__` is not needed at this location. The test I did for it was flawed. As long as there is no one referring to `Base_tsickle_CLosure_declarations` it will tree shake away.